core/trino-main/src/main/java/io/trino/execution/DropDefaultValueTask.java

```diff
@@ -0,0 +1,109 @@
+import static io.trino.spi.connector.ConnectorCapabilities.DEFAULT_COLUMN_VALUE;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static java.util.Objects.requireNonNull;
+
+public class DropDefaultValueTask
+        implements DataDefinitionTask<DropDefaultValue>
+{
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public DropDefaultValueTask(Metadata metadata, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "DROP DEFAULT";
+    }
+
+    @Override
+    public ListenableFuture<Void> execute(
+            DropDefaultValue statement,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
+    {
+        Session session = stateMachine.getSession();
+        QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTableName());
+        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, tableName);
+        if (redirectionAwareTableHandle.tableHandle().isEmpty()) {
+            String exceptionMessage = "Table '%s' does not exist".formatted(tableName);
+            if (metadata.getMaterializedView(session, tableName).isPresent()) {
+                exceptionMessage += ", but a materialized view with that name exists.";
+            }
+            else if (metadata.isView(session, tableName)) {
+                exceptionMessage += ", but a view with that name exists.";
+            }
+            if (!statement.isTableExists()) {
+                throw semanticException(TABLE_NOT_FOUND, statement, "%s", exceptionMessage);
+            }
+            return immediateVoidFuture();
+        }
+        accessControl.checkCanAlterColumn(session.toSecurityContext(), tableName);
+
+        TableHandle tableHandle = redirectionAwareTableHandle.tableHandle().get();
+        CatalogHandle catalogHandle = tableHandle.catalogHandle();
+        String column = statement.getColumnName().getValue();
+        ColumnHandle columnHandle = metadata.getColumnHandles(session, tableHandle).get(column);
+
+        if (columnHandle == null) {
+            throw semanticException(COLUMN_NOT_FOUND, statement, "Column '%s' does not exist", column);
+        }
+
+        ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle, columnHandle);
+        if (columnMetadata.isHidden()) {
+            throw semanticException(NOT_SUPPORTED, statement, "Cannot modify hidden column");
+        }
+        if (columnMetadata.getDefaultValue().isEmpty()) {
+            throw semanticException(NOT_SUPPORTED, statement, "Column '%s' does not have a default value", column);
```

Comment:
If this failure is required by the spec, we should use another error code.
`NOT_SUPPORTED` is for cases where we don't follow the spec yet.